wgengine/userspace.go

```diff
@@ -255,32 +255,27 @@ func newUserspaceEngineAdvanced(logf logger.Logf, tundev *tstun.TUN, routerGen R
 	return e, nil
 }
 
-// pinger sends ping packets for a few seconds.
-//
-// These generated packets are used to ensure we trigger the spray logic in
-// the magicsock package for NAT traversal.
-func (e *userspaceEngine) pinger(peerKey wgcfg.Key, ips []wgcfg.IP) {
-	e.logf("generating initial ping traffic to %s (%v)", peerKey.ShortString(), ips)
-	var srcIP packet.IP
+type pinger struct {
+	e      *userspaceEngine
+	done   chan struct{}
+	cancel context.CancelFunc
+}
 
-	e.wgLock.Lock()
-	if len(e.lastCfg.Addresses) > 0 {
-		srcIP = packet.NewIP(e.lastCfg.Addresses[0].IP.IP())
-	}
-	e.wgLock.Unlock()
+func (p *pinger) close() {
```

Comment:
Document that close() cannot be called while holding e.mu. That seems like an easy deadlock trap.